nixos/modules/programs/firefox.nix

```diff
@@ -81,14 +81,38 @@ in
       ];
     };
 
+    finalPackage = lib.mkOption {
+      description = "Final package (read-only)";
+      type = lib.types.package;
+      readOnly = true;
+      default = cfg.package.override (old: {
+        extraPrefsFiles =
+          old.extraPrefsFiles or [ ]
+          ++ cfg.autoConfigFiles
+          ++ [ (pkgs.writeText "firefox-autoconfig.js" cfg.autoConfig) ];
+        nativeMessagingHosts = old.nativeMessagingHosts or [ ] ++ cfg.nativeMessagingHosts.packages;
+        cfg = (old.cfg or { }) // cfg.wrapperConfig;
+      });
+      defaultText = lib.literalExpression ''
+        cfg.package.override (old: {
+          extraPrefsFiles =
+            old.extraPrefsFiles or [ ]
+            ++ cfg.autoConfigFiles
+            ++ [ (pkgs.writeText "firefox-autoconfig.js" cfg.autoConfig) ];
+          nativeMessagingHosts = old.nativeMessagingHosts or [ ] ++ cfg.nativeMessagingHosts.packages;
+          cfg = (old.cfg or { }) // cfg.wrapperConfig;
+        });
+      '';
```

Comment:
I think either in the `description` or instead or a repetition of the code (if I want the code, I look at the sources, no?) as `default`, there should be a prosa description of what the `finalPackage` is. E.g.:

> The input `.package` with `.autoConfig`, `.nativeMessagingHosts`, `.wrapperConfig`, etc, applied to it.